src/file-events.c

```diff
@@ -139,13 +140,14 @@ static __always_inline void enter_mkdir(struct pt_regs *ctx)
     return;
 }
 
-static __always_inline void store_dentry(struct pt_regs *ctx, void *dentry)
+static __always_inline void store_dentry(struct pt_regs *ctx, void *path, void *dentry)
 {
     u64 pid_tgid = bpf_get_current_pid_tgid();
 
     load_event(incomplete_mkdirs, pid_tgid, incomplete_mkdir_t);
     if (event.target_dentry == NULL) {
         event.target_dentry = dentry;
```

Comment:
should we early guard against this being NULL? I know we check for it being NULL later but I am wondering if we should be checking it before putting it into the `incomplete_mkdirs` at all.

Reply:
The kernel has just checked that the dentry is valid before calling this; I don't think it's worth checking again for a pointer directly passed to us rather than something where we have to find a field via an offset.